File: core/src/test/java/io/pivotal/main/TestAdaptersModule.java
package io.pivotal.main;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class TestAdaptersModule {

    @Provides
    @Singleton
    ItemRepository provideItemRepository() {
        return new InMemoryItemRepository();
    }
}

File: core/src/test/java/com/billiamram/main/TestAdaptersModule.kt
package com.billiamram.main

import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class TestAdaptersModule {

    @Provides
    @Singleton
    internal fun provideItemRepository(): ExampleModelRepository {
        return InMemoryExampleModelRepository()
    }
}
